src/main/plugin/iso19139.ca.HNAP/schematron/schematron-rules-common.sch

```diff
@@ -403,6 +429,22 @@
 
     </sch:rule>
 
+    <sch:rule context="//gmd:distributionInfo/gmd:MD_Distribution/gmd:transferOptions/gmd:MD_DigitalTransferOptions/gmd:onLine/gmd:CI_OnlineResource">
+        <sch:let name="locLabel" value="document(concat('../loc/', $lang, '/labels.xml'))"/>
+        <sch:let name="protocolList" value="$locLabel/labels/element[@name='gmd:protocol']/helper/option"/>
+        <sch:let name="protocol" value="gmd:protocol/gco:CharacterString/text()"/>
+        <sch:let name="isValidProtocol" value="$protocol = $protocolList"/>
+
+    		<sch:let name="resourceName" value="gmd:name/gco:CharacterString/text()" />
+
+    		<sch:let name="protocolListString" value="geonet:protocolListString($protocolList)"/>
+
+    		<sch:let name="protocolListMsg" value="geonet:appendLocaleMessage($loc/strings/OnlineResourceProtocol, $protocolListString)"/>
+    		<sch:let name="locMsg" value="geonet:appendLocaleMessage($resourceNameMsg, $protocolListMsg)"/>
```

Comment:
@wangf1122 where is defined `$resourceNameMsg`? Seem it's missing in the PR, we just notice doing further testing.